cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -229,6 +231,10 @@ public SparkClassRewriter(Function<String, InputStream> resourceLookup, boolean
       // Rewrite the Spark OutputMetrics to skip overwriting bytes written metrics with 0.
       return rewriteOutputMetrics(SPARK_OUTPUT_METRICS, input);
     }
+    if (className.startsWith(CHECKPOINT_WRITE_HANDLER_TYPE.getClassName())) {
```

Comment:
This should be an optional rewrite controlled by cdap-site.xml

Reply:
@chtyim Should this feature be enabled or disabled by default?

Reply:
disable